pkg/sql/create_index.go

```diff
@@ -203,6 +208,16 @@ func validateIndexColumnsExist(
 	return nil
 }
 
+// validateIndexNameIsUnique validates that the index name does not exist.
+func validateIndexNameIsUnique(desc *sqlbase.MutableTableDescriptor, indexName string) error {
+	for _, index := range desc.AllNonDropIndexes() {
```

Comment:
Because we do non-drop here, couldn't the following happen:
* We start an index drop.
* We create a new index with the same name
* we rollback the drop job
Now we have two indexes with the same name?